src/pymedphys/dcm/_level1/dcmdose.py

```diff
@@ -41,12 +40,33 @@
 from ...libutils import get_imports
 IMPORTS = get_imports(globals())
 
-Coords = namedtuple('Coords', 'x y z')
-Dose = namedtuple(
-    'Dose', 'values units type summation heterogeneity_correction')
-
 # pylint: disable=C0103
 
+class DicomDose:
+
+    def __init__(self, dcm_filepath):
+        dcm = pydicom.dcmread(dcm_filepath)
+        if dcm.Modality != "RTDOSE":
+            raise ValueError("The input DICOM file is not an RT Dose file")
+        self.dcm = dcm
+
+        self.values = dcm.pixel_array * dcm.DoseGridScaling
+        self.units = dcm.DoseUnits
+        self.x, self.y, self.z = extract_dicom_patient_xyz(dcm)
+        self.coords = convert_xyz_to_dicom_coords((self.x, self.y, self.z))
+        self.mask = None
```

Comment:
Meant for masking dose by an ROI from a DICOM RT Structure File. Code (eventually) forthcoming.

Reply:
Some of the code to achieve this should be able to be found at https://github.com/pymedphys/pymedphys/blob/master/src/pymedphys/dcm/_level1/dcmdose.py#L654-L686

Reply:
Yep, you're absolutely right - I had it in mind while writing `convert_xyz_to_dicom_coords()`.